File: app/src/main/java/com/xwm/androidproject/ui/MainActivity.java
package com.xwm.androidproject.ui;

import android.os.Bundle;
import android.widget.TextView;

import com.xwm.androidproject.R;
import com.xwm.androidproject.base.BaseActivity;

public class MainActivity extends BaseActivity {
    private TextView mTxtTest;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mTxtTest = findViewById(R.id.tv_test);
    }
}

File: app/src/main/java/com/xwm/androidproject/ui/MainActivity.kt
package com.xwm.androidproject.ui

import com.xwm.androidproject.R
import com.xwm.androidproject.databinding.ActivityMainBinding
import com.xwm.base.base.BaseActivity


class MainActivity : BaseActivity<MainViewModel, ActivityMainBinding>() {

    override fun initView() {
        mDataBinding.btnTest.setOnClickListener {
            test()
        }
    }

    override fun initDataObserver() {
        mViewModel.nicknameData.observe(this, {
            mDataBinding.btnTest.text = it
        })
    }

    override fun loadData() {

    }

    private fun test() {
        mViewModel.getNickname()
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

}